base_rest/restapi.py

```diff
@@ -0,0 +1,250 @@
+    :param auth: The type of authentication method, can on of the following:
+    :param cors: The Access-Control-Allow-Origin cors directive value.
+    :param bool csrf: Whether CSRF protection should be enabled for the route.
+                      Defaults to ``False``
+
+    """
+
+    def decorator(f):
+        _routes = []
+        for paths, http_method in routes:
+            if not isinstance(paths, list):
+                paths = [paths]
+            _routes.append(([p for p in paths], http_method))
+        routing = {
+            "csrf": csrf,
+            "auth": auth,
+            "cors": cors,
+            "routes": _routes,
+            "input_param": input_param,
+            "output_param": output_param,
+        }
+
+        @functools.wraps(f)
+        def response_wrap(*args, **kw):
+            response = f(*args, **kw)
+            return response
+
+        response_wrap.routing = routing
+        response_wrap.original_func = f
+        return response_wrap
+
+    return decorator
+
+
+class RestMethodParam(object):
+    def from_params(self, service, params):
+        """
+        This method is called to process the parameters received at the
+        controller. This method should validate and sanitize these paramaters.
+        It could also be used to transform these parameters into the format
+        expected by the called method
+        :param service:
+        :param request: `HttpRequest.params`
+        :return: Value into the format expected by the method
+        """
+        pass
+
+    def to_response(self, service, result):
+        """
+        This method is called to prepare the result of the call to the method
+        in a format suitable by the controller (http.Response or JSON dict).
+        It's responsible for validating and sanitizing the result.
+        :param service:
+        :param obj:
+        :return: http.Response or JSON dict
+        """
+        pass
+
+    def to_openapi_query_parameters(self, service):
+        return {}
+
+    def to_openapi_requestbody(self, service):
+        return {}
+
+    def to_openapi_response(self, service):
+        return {}
+
+
+class CerberusValidator(RestMethodParam):
+    def __init__(self, schema=None, is_list=False):
+        """
+
+        :param schema: can be dict as cerberus schema, an instance of
+                       cerberus.Validator or a sting with the method name to
+                       call on the service to get the schema or the validator
+        :param is_list:
+        """
+        self._schema = schema
+        self._is_list = is_list
+
+    def from_params(self, service, params):
+        validator = self.get_cerberus_validator(service)
+        if validator.validate(params):
+            return validator.document
+        raise UserError(_("BadRequest %s") % validator.errors)
+
+    def to_response(self, service, result):
+        validator = self.get_cerberus_validator(service)
+        if validator.validate(result):
```

Comment:
Not covered what will happen if it is a list. I have tested with:
```
    def to_response(self, service, result):
        validator = self.get_cerberus_validator(service)
        if self._is_list:
            data = []
            for res in result:
                if not validator.validate(res):
                    raise SystemError(_("Invalid Response %s") % validator.errors)
                data.append(validator.document)
            return data
        if validator.validate(result):
            return validator.document
        raise SystemError(_("Invalid Response %s") % validator.errors)
```
And worked perfectly

Reply:
@etobella Are you sure you use the right version of marshmallow_object and marshamallow? AFAIK is_list is natively supported by the marshmallow_object.... I've to check since I'm pretty sure that should work as is..

Reply:
It didn't work on my side. I will check again just to be sure